device/iot-device-samples/pnp-device-sample/Thermostat/src/main/java/samples/com/microsoft/azure/sdk/iot/Thermostat.java

```diff
@@ -0,0 +1,317 @@
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+@Slf4j
+public class Thermostat {
+
+    public enum StatusCode {
+        COMPLETED (200),
+        IN_PROGRESS (202),
+        NOT_FOUND (404);
+
+        private final int value;
+        StatusCode(int value) {
+            this.value = value;
+        }
+    }
+
+    private static final String deviceConnectionString = System.getenv("IOTHUB_DEVICE_CONNECTION_STRING");
+    private static final String MODEL_ID = "dtmi:com:example:Thermostat;1";
+
+    // Plug and play features are available over either MQTT or MQTT_WS.
+    private static final IotHubClientProtocol protocol = IotHubClientProtocol.MQTT;
+
+    private static final Random random = new Random();
+    private static final Map<Date, Double> temperatureReadings = new HashMap<>();
+
+    private static DeviceClient deviceClient;
+    private static double temperature = 0.0d;
+    private static double maxTemperature = 0.0d;
+    private static boolean temperatureReset = true;
+
+    public static void main(String[] args) throws URISyntaxException, IOException {
+
+        // This sample follows the following workflow:
+        // -> Initialize device client instance.
+        // -> Set handler to receive "targetTemperature" updates, and send the received update over reported property.
+        // -> Set handler to receive "getMaxMinReport" command, and send the generated report as command response.
+        // -> Periodically send "temperature" over telemetry.
+        // -> Send "maxTempSinceLastReboot" over property update, when a new max temperature is set.
+
+        log.debug("Initialize the device client.");
+        initializeDeviceClient();
+
+        log.debug("Start twin and set handler to receive \"targetTemperature\" updates.");
+        deviceClient.startDeviceTwin(new TwinIotHubEventCallback(), null, new TargetTemperatureUpdateCallback(), null);
+        Map<Property, Pair<TwinPropertyCallBack, Object>> desiredPropertyUpdateCallback =
+                Collections.singletonMap(
+                        new Property("targetTemperature", null),
+                        new Pair<TwinPropertyCallBack, Object>(new TargetTemperatureUpdateCallback(), null));
+        deviceClient.subscribeToTwinDesiredProperties(desiredPropertyUpdateCallback);
+
+        log.debug("Set handler to receive \"getMaxMinReport\" command.");
+        String methodName = "getMaxMinReport";
+        deviceClient.subscribeToDeviceMethod(new GetMaxMinReportMethodCallback(), methodName, new MethodIotHubEventCallback(), methodName);
+
+        new Thread(new Runnable() {
+            @SneakyThrows(InterruptedException.class)
```

Comment:
`SneakyThrows` will not ignore, wrap or modify the actual checked exception thrown, but will instead prevent the compiler from complaining that the exception is not handled (either by declaring `throws` n the method signature, or by wrapping it in a try-catch block and rethrowing a `RuntimeException`.

From lombok documentation: https://projectlombok.org/features/SneakyThrows
whatever exception propagates out of your run() method, checked or not, it will be passed to the Thread's unhandled exception handler. Catching a checked exception and wrapping it in some sort of RuntimeException is only obscuring the real cause of the issue.